cmd/launcher/svc_config_windows.go

```diff
@@ -0,0 +1,114 @@
+	if opts.KolideServerURL != "k2device.kolide.com" && opts.KolideServerURL != "k2device-preprod.kolide.com" {
+		return
+	}
+
+	// Get launcher service key
+	launcherServiceKey, err := registry.OpenKey(registry.LOCAL_MACHINE, launcherServiceRegistryKeyName, registry.ALL_ACCESS)
+	if err != nil {
+		level.Error(logger).Log("msg", "could not open registry key", "key_name", launcherServiceRegistryKeyName, "err", err)
+		return
+	}
+
+	// Close it once we're done
+	defer func() {
+		if err := launcherServiceKey.Close(); err != nil {
+			level.Error(logger).Log("msg", "could not close registry key", "key_name", launcherServiceRegistryKeyName, "err", err)
+		}
+	}()
+
+	// Check to see if we need to turn off delayed autostart
+	checkDelayedAutostart(launcherServiceKey, logger)
+
+	// Check to see if we need to update the service to depend on Dnscache
+	checkDependOnService(launcherServiceKey, logger)
+}
+
+// checkDelayedAutostart checks the current value of `DelayedAutostart` (whether to wait ~2 minutes
+// before starting the launcher service) and updates it if necessary.
+func checkDelayedAutostart(launcherServiceKey registry.Key, logger log.Logger) {
+	currentDelayedAutostart, _, getDelayedAutostartErr := launcherServiceKey.GetIntegerValue(delayedAutostartName)
+
+	// Can't determine current value, don't update
+	if getDelayedAutostartErr != nil {
+		return
+	}
+
+	// Delayed autostart is already disabled
+	if currentDelayedAutostart == uint64(delayedAutostartDisabled) {
+		return
+	}
+
+	// Turn off delayed autostart
+	if err := launcherServiceKey.SetDWordValue(delayedAutostartName, delayedAutostartDisabled); err != nil {
+		level.Error(logger).Log("msg", "could not turn off DelayedAutostart", "err", err)
+	}
+}
+
+// checkDependOnService checks the current value of `DependOnService` (the list of services that must
+// start before launcher can) and updates it if necessary.
+func checkDependOnService(launcherServiceKey registry.Key, logger log.Logger) {
+	serviceList, _, getServiceListErr := launcherServiceKey.GetStringsValue(dependOnServiceName)
+
+	if getServiceListErr != nil {
+		if getServiceListErr.Error() == notFoundInRegistryError {
+			// `DependOnService` does not exist for this service yet -- we can safely set it to include the Dnscache service.
+			if err := launcherServiceKey.SetStringsValue(dependOnServiceName, []string{dnscacheService}); err != nil {
+				level.Error(logger).Log("msg", "could not set strings value for DependOnService", "err", err)
+			}
+			return
+		}
+
+		// In any other case, if we can't get the current value, we don't want to proceed --
+		// we don't want to wipe any current data from the list.
+		return
+	}
+
+	// Check whether the service configuration already includes Dnscache in the list of services it depends on
+	foundDnscacheInList := false
+	for _, service := range serviceList {
+		if service == dnscacheService {
+			foundDnscacheInList = true
+			break
```

Comment:
You can early return the whole function, right?

Reply:
Yes, thank you! updated

Reply:
I <3 early returns